mesonbuild/environment.py

```diff
@@ -834,12 +834,23 @@ def detect_d_compiler(self, want_cross):
             raise EnvironmentException('Could not execute D compiler "%s"' % ' '.join(exelist))
         version = search_version(out)
         full_version = out.split('\n', 1)[0]
+
+        is_win64 = False
+        is_msvc = False
+        if mesonlib.is_windows():
+            is_msvc = 'VCINSTALLDIR' in os.environ
+            # detect which MSVC build environment is currently active
+            if is_msvc and detect_cpu_family({'c': self.detect_c_compiler(want_cross)}) == 'x86_64':
+                is_win64 = True
+            elif not is_msvc and detect_cpu_family({}) == 'x86_64':
+                is_win64 = True
```

Comment:
We already have code to do this detection, see: `detect_windows_arch()`. Does that not work?

Reply:
``detect_cpu_family`` calls the ``detect_windows_arch`` function, but I could call it directly too.